drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -376,3 +409,46 @@ int intel_adsp_hda_dma_pm_action(const struct device *dev, enum pm_device_action
 	return 0;
 }
 #endif
+
+#define DEVICE_DT_GET_AND_COMMA(node_id) DEVICE_DT_GET(node_id),
+
+void intel_adsp_hda_dma_isr(void)
+{
+#if CONFIG_DMA_INTEL_ADSP_HDA_TIMING_L1_EXIT
+	struct dma_context *dma_ctx;
+	const struct intel_adsp_hda_dma_cfg *cfg;
+	bool clear_l1_exit = false;
+	int i, j;
+	const struct device *host_dev[] = {
+#if CONFIG_DMA_INTEL_ADSP_HDA_HOST_OUT
+		DT_FOREACH_STATUS_OKAY(intel_adsp_hda_host_out, DEVICE_DT_GET_AND_COMMA)
+#endif
+#if CONFIG_DMA_INTEL_ADSP_HDA_HOST_IN
+		DT_FOREACH_STATUS_OKAY(intel_adsp_hda_host_in, DEVICE_DT_GET_AND_COMMA)
+#endif
+	};
+
+	for (i = 0; i < ARRAY_SIZE(host_dev); i++) {
+		dma_ctx = (struct dma_context *)host_dev[i]->data;
+		cfg = host_dev[i]->config;
+
+		for (j = 0; j < dma_ctx->dma_channels; j++) {
+			if (atomic_test_bit(dma_ctx->atomic, j)) {
+				clear_l1_exit |=
+					intel_adsp_hda_check_buffer_interrupt(cfg->base,
+									      cfg->regblock_size, j);
+				intel_adsp_hda_disable_buffer_interrupt(cfg->base,
+									cfg->regblock_size, j);
+				intel_adsp_hda_clear_buffer_interrupt(cfg->base,
+								      cfg->regblock_size, j);
+			}
+		}
+	}
+
+	if (clear_l1_exit) {
+#if CONFIG_SOC_SERIES_INTEL_ACE
+		ACE_DfPMCCH.svcfg &= ~(ADSP_FORCE_DECOUPLED_HDMA_L1_EXIT_BIT);
```

Comment:
An alternative approach would be to assign a register address and mask pair in DT perhaps to avoid tying this so close to INTEL_ACE and its semantics. Then in this code no ACE particulars would be needed, just DT attributes for l1 exit register updates perhaps.

Reply:
The problem is ACE_DfPMCCH is already coming from DT so probably we dont want to maintain the same registry in several DT fields?